### PR TITLE
gh-118868: logging QueueHandler fix passing of kwargs

### PR DESCRIPTION
**Pull request by Kaundur**
* Issue: gh-118868

### CODE REVIEW

File: Lib/test/test_logging.py [hunk 1]
```diff
@@ -3976,6 +3976,34 @@ def test_111615(self):
         }
         logging.config.dictConfig(config)
 
+    def test_118868(self):
```
Comment (nineteendo) [inline Lib/test/test_logging.py]:
```suggestion
    # gh-118868: check if kwargs are passed to logging QueueHandler
    def test_kwargs_passing(self):
```

File: Lib/logging/config.py [hunk 2]
```diff
@@ -725,16 +725,20 @@ def add_filters(self, filterer, filters):
 
     def _configure_queue_handler(self, klass, **kwargs):
         if 'queue' in kwargs:
-            q = kwargs['queue']
+            q = kwargs.pop('queue')
         else:
             q = queue.Queue()  # unbounded
         rhl = kwargs.get('respect_handler_level', False)
```
Comment (dolfinus) [inline Lib/logging/config.py]: Shouldn't there be `.pop()` as well?

Comment (Kaundur) [reply Lib/logging/config.py]: Yes you are correct, 'respect_handler_level' should have been included

File: Lib/logging/config.py [hunk 3]
```diff
@@ -725,16 +725,25 @@ def add_filters(self, filterer, filters):
...
-            q = kwargs['queue']
+            q = kwargs.pop('queue')
         else:
             q = queue.Queue()  # unbounded
-        rhl = kwargs.get('respect_handler_level', False)
+
+        if 'respect_handler_level' in kwargs:
+            rhl = kwargs.pop('respect_handler_level')
+        else:
+            rhl = False
```
Comment (vsajip) [inline Lib/logging/config.py]:
```suggestion
        rhl = kwargs.pop('respect_handler_level', False)
```

File: Lib/logging/config.py [hunk 4]
```diff
@@ -725,16 +725,25 @@ def add_filters(self, filterer, filters):
...
+
```
Comment (vsajip) [inline Lib/logging/config.py:735]: Use the same approach as suggested above: 
```python
lklass = kwargs.pop('listener', logging.handlers.QueueListener)
handlers = kwargs.pop(`handlers`, [])
```

**Review by vsajip — CHANGES_REQUESTED** [review]

**Comment by Kaundur** [comment]
I have made the requested changes; please review again

**Review by vsajip — APPROVED** [review]

**Pull request merged**